### DISCUSSION TITLE
NixOS port

### DISCUSSION THREAD

**Discussion by kopsis** [nwg-piotr/nwg-shell#302] [Porting for other distros]
A number of nwg-shell components have Nix packages. Most work, but a few (like nwg-look) have functional issues and several are still missing. I have fixed the problems in nwg-look and have preliminary working versions of nwg-shell and nwg-shell-config. Before I tackle the next one on my todo list (azote), I figured it would be worth checking to see if anyone else is actively working on full nwg-shell support on NixOS? If so, I'd like to explore how we can coordinate these efforts.

**Comment by nwg-piotr** [comment]
Hi David,

> if anyone else is actively working on full nwg-shell support on NixOS?

Nobody, as far as I know. Just in case, you could also ask on the [nwg-shell Matrix space](https://matrix.to/#/#nwg-shell:matrix.org).

**Comment by ShaddyDC** [comment]
I'm not working on it, but I'm curious to try it out, so I'm following this discussion here in hopes somebody makes a simple-to-use flake. You can find some other people's attempts with the github search: https://github.com/search?q=lang%3Anix+nwg-shell&type=code

**Reply by nwg-piotr** [reply]
Wish I could cooperate, but I know nothing on NixOS, and have no time to learn. I'm receiving 5 (on average) bug reports / feature requests daily, and I only have 3 spare hours a day.

**Reply by ShaddyDC** [reply]
I appreciate the sentiment! No pressure from my side though. If somebody does put in the work and is happy to share their solution, I'll be happy, but if not, I'll be fine as well. You're already creating what looks like an amazing project; I don't expect you to put in the work to support all kinds of environments that are just not relevant to you too :)

**Reply by kopsis** [reply]
I ultimately decided that for my use-case, the effort of doing and maintaining a port simply wasn't going to pay off. If you take things to the logical nixos conclusion of a fully declarative config, much of what nwg-shell does really well ends up unused.
